### PR TITLE
otelzap: Implement namespace method

### PR DESCRIPTION
**Pull request by khushijain21**
Part of https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5191

Pre-work https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5279

The approach is to use linked list to open a namespace and keep track of it.
`calculate` method iterates to the most nested namespace and appends data in reverse order.

### CODE REVIEW

**Comment by khushijain21** [comment]
The approach is to use linked list to open a namespace and keep track of it.

`getObjValue` method iterates to the most nested namespace and appends data in reverse order.

**Comment by pellared** [comment]
`build_and_test / lint (pull_request)` is failing. Can you fix it before I make a review?

File: bridges/otelzap/encoder.go [hunk 1]
```diff
@@ -18,47 +18,69 @@ var (
 	_ zapcore.ArrayEncoder  = (*arrayEncoder)(nil)
 )
 
+type newNameSpace struct {
+	ns   string
+	kv   []log.KeyValue
```
Comment (pellared) [inline bridges/otelzap/encoder.go:23]: Can we rename it to "attrs" (for attributes)?

```suggestion
	attrs []log.KeyValue
```

File: bridges/otelzap/encoder.go [hunk 2]
```diff
@@ -18,47 +18,69 @@ var (
...
 	}
+	return &objectEncoder{
+		root: m,
+		cur:  m,
+	}
+}
+
+// It iterates to the end of the linked list and appends namespace data.
+// Run this function before accessing complete result.
+func (m *objectEncoder) getEncodedResult(o *newNameSpace) {
```
Comment (pellared) [inline bridges/otelzap/encoder.go:49]: Can we rename it to something like `calculate` or `aggregate`? The current name suggests that it returns something and that it is a pure function (accessor)

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**